File: src/main/kotlin/tech/diggle/apps/accomodation/candidates/CandidateRepository.java
package tech.diggle.apps.accomodation.candidates;


import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;


@Repository

public interface CandidateRepository extends JpaRepository<Candidate, Long> {
    Candidate findByCandidateName(String candidateName);

    Candidate findByParty(String party);
}

File: src/main/kotlin/tech/diggle/apps/accomodation/candidate/CandidateRepository.kt
package tech.diggle.apps.accomodation.candidate

import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository

interface CandidateRepository : PagingAndSortingRepository<Candidate, Long> {
    fun findByElectionId(id: Long, pageRequest: Pageable): Page<Candidate>
    fun findByFirstNameAndLastName(firstName: String, lastName: String): Candidate?
}